File: java/program/Toko.java
// Class Toko (Parent)
public class Toko {
    private String id;
    private String namaProduk;
    private int harga;

    // Constructor
    public Toko() {
        this.id = "";
        this.namaProduk = "";
        this.harga = 0;
    }

    public Toko(String id, String namaProduk, int harga) {
        this.id = id;
        this.namaProduk = namaProduk;
        this.harga = harga;
    }

    // Getter & Setter
    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public String getNamaProduk() {
        return namaProduk;
    }

    public void setNamaProduk(String namaProduk) {
        this.namaProduk = namaProduk;
    }

    public int getHarga() {
        return harga;
    }

    public void setHarga(int harga) {
        this.harga = harga;
    }
}

File: cpp/program/Toko.cpp
#include <iostream>
using namespace std;

// Class Toko (Parent)
class Toko {
private: 
    string id;          // Atribut id
    string namaProduk;  // Atribut nama Produk
    int harga;          // Atribut harga

public: 
    // Constructor kosong
    Toko() { 
    }

    // Constructor dengan parameter
    Toko(string id, string namaProduk, int harga) {
        this->id = id;
        this->namaProduk = namaProduk;
        this->harga = harga;
    }

    // Getter setter 
    string getId() const {
        return id;
    }
    void setId(string id) {
        this->id = id;
    }

    string getNamaProduk() const {
        return namaProduk;
    }
    void setNamaProduk(string namaProduk) {
        this->namaProduk = namaProduk;
    }

    int getHarga() const {
        return harga;
    }
    void setHarga(int harga) {
        this->harga = harga;
    }
};
